src/Jackett.Common/Indexers/AniLibria.cs

```diff
@@ -0,0 +1,189 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using AngleSharp.Dom;
+using AngleSharp.Html.Parser;
+using AngleSharp.Html.Dom;
+using Jackett.Common.Models;
+using Jackett.Common.Models.IndexerConfig;
+using Jackett.Common.Services.Interfaces;
+using Jackett.Common.Utils;
+using Jackett.Common.Utils.Clients;
+using Newtonsoft.Json.Linq;
+using NLog;
+using System.Text.RegularExpressions;
+
+namespace Jackett.Common.Indexers
+{
+    [ExcludeFromCodeCoverage]
+    public class AniLibria : BaseWebIndexer
+    {
+        public AniLibria(IIndexerConfigurationService configService, Utils.Clients.WebClient wc, Logger l, IProtectionService ps)
+            : base(id: "AniLibria",
+                   name: "AniLibria",
+                   description: "AniLibria is a Public torrent tracker for anime, voiced on russian by AniLibria team",
+                   link: "https://www.anilibria.tv/",
+                   caps: new TorznabCapabilities(),
+                   configService: configService,
+                   client: wc,
+                   logger: l,
+                   p: ps,
+                   configData: new ConfigurationData())
+        {
+            Encoding = Encoding.UTF8;
+            Language = "ru-ru";
+            Type = "public";
+
+            // Configure the category mappings
+            AddCategoryMapping(1, TorznabCatType.TVAnime, "Anime");
+        }
+
+        private Uri GetAbsoluteUrl(string relativeUrl) => new Uri(SiteLink + Regex.Replace(relativeUrl, @"\\/", "/").TrimStart('/'));
+
+        /// <summary>
+        /// https://www.anilibria.tv/public/search.php
+        /// </summary>
+        private Uri SearchUrl => GetAbsoluteUrl("/public/search.php");
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+            var releases = await PerformQuery(new TorznabQuery());
+
+            await ConfigureIfOK(string.Empty, true, () =>
+                throw new Exception("Could not find releases from this URL"));
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            // Prepare the search query
+            var queryParameters = new List<KeyValuePair<string, string>>
+            {
+                { "search", query.SearchTerm },
```

Comment:
If you type "Harry S01E02" in Jackett search box:
* query.SearchTerm = Harry
* query.GetQueryString() = Harry S01E02
I don't know if this site supports episode search. Use the most suitable

What happens if query.SearchTerm = "" ? Does the site return latest torrents?
If search=<empty> works that's it. If not you have to add some logic to perform different searches for latest releases or search with term. Example => https://github.com/Jackett/Jackett/blob/621a47338dcd98ff0b2471b7df99f7995e10a4dd/src/Jackett.Common/Indexers/AwesomeHD.cs#L82

Reply:
This site returns nothing if search request is empty.
Also it's not able to search by episodes, cause each page corresponds to some release(show) and on this page we have different torrent files (with different episode ranges and quality options).

About latest - i have no idea how to implement it right now.
Search isn't able to return all releases.
It also doesn't seems that site has got some news feed about releases.
All I have found - is https://www.anilibria.tv/pages/catalog.php where we can get list of all releases. But it's hard to organise that as a history, cause we need to process all pages, all torrents on that pages and only here we would be able to get publication date.

So maybe you have some advices for such situation?

Reply:
@garfield69 ping

Reply:
It seems to me that the web site is not suitable for use by scraping html results via an indexer.
the presentation structure and search engine option on that web site do not appear to be in the typical torrent tracker design.
 
however anilibria does appear to have an API, so perhaps you could read the docs and see if the api queries support what we need to have a functioning indexer.
https://github.com/anilibria/docs/blob/master/api_v2.md
the indexer needs to achieve two main features:
1. in response to a keywordless search from the user the indexer should present either:
  - a list of between 10 to 100 torrents sorted by the most recent publishing dates
  - or a list of any 10 to 100 torrents published this year
  - or at least 1 torrent so that the indexer can show that the api server is online and returning results
2. in response to a keyword search (such as: `black clover 73 1080p`) the indexer should present a list of torrents that best match those terms

if you cannot find in the docs the api commands to deliver these two requirements then we close 5762 and call it quits.

Reply:
So, as far as i understand, second "main feature" for now works pretty well now, i think.

And thank you for the api info, cause it seems that https://github.com/anilibria/docs/blob/master/api_v2.md#%D0%B2%D1%81%D0%B5-%D0%B4%D0%BE%D1%81%D1%82%D1%83%D0%BF%D0%BD%D1%8B%D0%B5-%D0%BF%D0%B0%D1%80%D0%B0%D0%BC%D0%B5%D1%82%D1%80%D1%8B-7 could help us with #1 feature.

I'll take a closer look on this API

Reply:
Added fetching of releases history via API